### DISCUSSION TITLE
Fetch options. What is the priority/outcome of combining revalidate AND tags?

### DISCUSSION THREAD

**Discussion by james2doyle** [vercel/next.js#70355] [Help] [answered]
### Summary

Anyone know what happens when you use using the `fetch` cache controls for [options.next.revalidate](https://nextjs.org/docs/app/building-your-application/caching#fetch-optionsnextrevalidate) and [options.next.tags](https://nextjs.org/docs/app/building-your-application/caching#fetch-optionsnexttags-and-revalidatetag) together with ISR?

Will revalidate be ignored when a tag is being used or will revalidate be honored, and the freshest data will be based on that?

**Comment by icyJoseph** [comment, marked as answer]
Hi,

As far as I can see, the revalidate by time, and by tags, should not interfere with each other. So yeah, you can use both.

I looked at the implementation of these, and then ran a couple of tests in production mode.

```tsx
export default async function Home() {
  const num = await fetch(
    "https://www.random.org/integers/?num=1&min=1&max=1000&col=1&base=10&format=plain&rnd=new",
    {
      next: {
        revalidate: 60,
        tags: ["rng"],
      },
    },
  ).then((res) => res.text());

  return <div>Hello {num}</div>;
}
```

And this revalidate route:

```ts
import { revalidateTag } from "next/cache";

export const dynamic = "force-dynamic";

export async function GET() {
  revalidateTag("rng");
  return Response.json({ revalidated: true, now: Date.now() });
}
```

After building, I go to the home page, and see a number, then I can revalidate the tag running this on my browser console:

```js
fetch("/tag").then(res => res.json()).then(console.log)
```

And reloading the page, shows a new number (assuming you don't get the same random number...), and then waiting 60 seconds, to then do a `double` request (as per documentation, that's how stale-while-revalidate, works), shows a new number, so these work fine together.

**Reply by james2doyle** [reply]
Thanks for the detailed response!

**Reply by icyJoseph** [reply]
Please don't forget to mark as answered, if you are satisfied with the answer.
